src/meson.build

```diff
@@ -60,10 +60,28 @@ pari = cc.find_library('pari')
 
 mpfr = cc.find_library('mpfr')
 
-# Cannot be found by pkg-config
-flint = cc.find_library('flint')
+# Once https://github.com/mesonbuild/meson/pull/12616 is released, we can simply do the following:
+# flint = dependency('flint', version : '>=3.0.0')
+flint = dependency('flint', version : '>=3.0.0', required: false, allow_fallback: false)
+if not flint.found()
+    flint = subproject('flint').get_variable('flint_dep')
```

Comment:
I don't think this is right.. should it just be falling back to `cc.find_library('flint')` if the `dependency()` call failed?

Currently it's trying to download flint on a machine where it's already installed...

Reply:
If you have flint 2.* installed, the compilation will not succeed as we recently broke backwards compatiblity. So in order to compile sage on recentish systems, I had to add this fallback (see https://github.com/sagemath/sage/pull/37077 for the tests). (Compilation works, but there are now runtime errors related to arb/flint - haven't found the time to investigate this further.)

Reply:
Nevermind, I assumed that I had a working pkg-config file for my flint-3.0.1, but it looks like it isn't installed by default. One more thing to investigate.

Reply:
For the record: https://github.com/flintlib/flint/commit/c37bb5b000b2988314134c57e27b9d6423434a6e

Reply:
The pkg-config of flint 3.0 is also not working properly, you need https://github.com/flintlib/flint/pull/1647.

Reply:
Ah, thanks, but it's installing the `*.pc` file to the wrong location anyway.